exercises/practice/zebra-puzzle/.meta/src/ZebraPuzzle.example.elm

```diff
@@ -0,0 +1,321 @@
+module ZebraPuzzle exposing (Nationality(..), drinksWater, ownsZebra)
+
+
+type Number
+    = One
+    | Two
+    | Three
+    | Four
+    | Five
+
+
+type Nationality
+    = Japanese
+    | Norwegian
+    | English
+    | Spanish
+    | Ukrainian
+
+
+type Color
+    = Red
+    | Green
+    | Blue
+    | Yellow
+    | Ivory
+
+
+type Drink
+    = Tea
+    | Milk
+    | OrangeJuice
+    | Coffee
+    | Water
+
+
+type Animal
+    = Zebra
+    | Dog
+    | Snails
+    | Horse
+    | Fox
+
+
+type CigaretteBrand
+    = Chesterfields
+    | OldGold
+    | Kools
+    | LuckyStrike
+    | Parliaments
+
+
+type alias House =
+    { number : Number
+    , nationality : Nationality
+    , color : Color
+    , drink : Drink
+    , animal : Animal
+    , cigaretteBrand : CigaretteBrand
+    }
+
+
+drinksWater : Maybe Nationality
+drinksWater =
+    solution
+        |> List.filter (\house -> .drink house == Water)
+        |> List.map .nationality
+        |> List.head
+
+
+ownsZebra : Maybe Nationality
+ownsZebra =
+    solution
+        |> List.filter (\house -> .animal house == Zebra)
+        |> List.map .nationality
+        |> List.head
+
+
+solution : List House
+solution =
+    let
+        reducedPossibilities =
+            allPossibilities |> applySameHouseConstraints sameHouseConstraints |> applyneighboringHouseConstraint
+
+        iterate possibilities =
+            let
+                newPossibilities =
+                    possibilities |> applySameHouseConstraints (findNewSameHouseConstraints possibilities) |> applyneighboringHouseConstraint
+            in
+            if List.length newPossibilities == List.length possibilities then
+                possibilities
+
+            else
+                iterate newPossibilities
+    in
+    iterate reducedPossibilities
+
+
+allPossibilities : List House
+allPossibilities =
```

Comment:
Is there something in List.Extra that makes this easier?

Reply:
Mmh, yeah maybe `cartesianProduct` + `transpose` + `List.map5` could make it cleaner. `unique` is also in there.
But we don't have access to `List.Extra` with the standard `elm.json`.

Reply:
If its a lot easier, then its probably worth a comment at least, so that a student will know there is a better way, even if we don't do it in the exemplar.

Reply:
I don't think it's that much easier, even with the functions I suggested you would need a messy pattern match for a list of 5 elements... And using another package will mean the test-runner and the analyzer will not run, so I think it's a suggestion with potential negative consequences. 
 
The way I did it looks messy but it's not complex, it's the same thing repeated 5 times. It's probably fine.